wrappers/services.go

```diff
@@ -705,41 +706,160 @@ func makeAbbrevWeekdays(start time.Weekday, end time.Weekday) []string {
+			// Monday; Monday happens during the 7 days,
+			// Tuesday can possibly happen on the 8th day if
+			// the month started on Tuesday)
+			//
+			// mon-tue1 - *-*~1 *-*-1..7 (anchored at first
+			// Tuesday; matching Monday can happen on the
+			// last day of previous month if Tuesday is the
+			// 1st)
+			//
+			// mon5-tue - *-*~1..7 *-*-1 (anchored at last
+			// Monday, the matching Tuesday can still happen
+			// within the last 7 days, or on the 1st of the
+			// next month)
+			//
+			// fri4-mon - *-*-22-31 *-*-1..7 (anchored at 4th
+			// Friday, can span onto the next month, extreme case in
+			// February when 28th is Friday)
+			//
+			// XXX: since old versions of systemd, eg. 229 available
+			// in 16.04 does not support x..y ranges, days need to
+			// be enumerated like so:
+			// Mon *-*-1..7 -> Mon *-*-1,2,3,4,5,6,7
+			//
+			// XXX: old systemd versions do not suppor the last n
+			// days syntax eg, *-*~1, thus the range needs to be
+			// generated in more verbose way like so:
+			// Mon *-*~1..7 -> Mon *-*-21,22,23,24,25,26,27,28,29,30,31
+			// (21-28 is the last week, but the month can have
+			// anywhere from 28 to 31 days)
+			//
+			startPos := week.Start.Pos
+			endPos := startPos
+			if !week.AnchoredAtStart() {
+				startPos = week.End.Pos
+				endPos = startPos
+			}
+			startDay := (startPos-1)*7 + 1
+			endDay := (endPos) * 7
+
+			if week.IsSingleDay() {
+				// single day, can use the 'weekday' filter
+				if startPos == timeutil.LastWeek {
+					// last week of a month, which can be
+					// 21-28 in case of February, while
+					// month can have between 28 and 31 days
 					days = append(days,
-						fmt.Sprintf("*-*-%d..%d/1", startDay, endDay))
+						fmt.Sprintf("%s *-*-%s", abbrev, daysRange(22, 31)))
 				} else {
 					days = append(days,
-						fmt.Sprintf("%s *-*-%d..%d/1", abbrev, startDay, endDay))
+						fmt.Sprintf("%s *-*-%s", abbrev, daysRange(startDay, endDay)))
 				}
+				continue
+			}
 
+			if week.AnchoredAtStart() {
+				// explore the edge cases first
+				switch startPos {
+				case timeutil.LastWeek:
+					// starts in the last week of the month and
+					// possibly spans into the first week of the
+					// next month;
+					// month can have between 28 and 31
+					// days
+					days = append(days,
+						// trailing 29-31 that are not part of a full week
+						fmt.Sprintf("*-*-%s", daysRange(29, 31)),
+						fmt.Sprintf("*-*-%s", daysRange(1, 7)))
+				case 4:
+					// a range in the 4th week can span onto
+					// the next week, which is either 28-31
+					// or in extreme case (eg. February with
+					// 28 days) 1-7 of the next month
+					days = append(days,
+						// trailing 29-31 that are not part of a full week
+						fmt.Sprintf("*-*-%s", daysRange(29, 31)),
+						fmt.Sprintf("*-*-%s", daysRange(1, 7)))
+				default:
+					// can possibly spill into the next week
+					days = append(days,
+						fmt.Sprintf("*-*-%s", daysRange(startDay+7, endDay+7)))
```

Comment:
Why do we advance by 7 days here (just confused, a little more explanation needed, please bear with me)?

Reply:
Consider the schedule `mon1-fri` and a calendars below:
```
    January 2019    
Su Mo Tu We Th Fr Sa
       1  2  3  4  5
 6  7  8  9 10 11 12
13 14 15 16 17 18 19
20 21 22 23 24 25 26
27 28 29 30 31 
```
In this case mon1 is 07.01 and the schedule ends on 11.01. For comparison:
```
     April 2019     
Su Mo Tu We Th Fr Sa
    1  2  3  4  5  6
 7  8  9 10 11 12 13
14 15 16 17 18 19 20
21 22 23 24 25 26 27
28 29 30
```
The schedule starts on 01.04 and ends on 05.04. 

In order to cover all the cases, we need to extend the calendar events to the next week (or previous week if the schedule is anchored at the end).

Reply:
Ah, I see it now, thanks for extra explanation!